### ISSUE TITLE
migrate bloomfilter foreachBatch from separate writer to the main writer's foreachBatch

### ISSUE THREAD

**Issue by kortemik** [teragrep/pth_10#186]
**Description**
currently bloomfilters are generated in their own writeStream step. This causes main writeStream step to re-materialize the query, meaning that the data is read two times. Further more it hides the progress while bloomfilters are being generated.

This could be solved by separating the aggregation phase (bloomfilter aggregator) and the save so that aggregation happens while the data is still in parallel mode and save after it is in sequential instead of the current implementation where a new outputStream is created and a foreach batch is applied on top of it.

**Comment by elliVM** [comment]
bloomfilter aggregation created as own teragrep step in its own node and run in parallel mode before bloom create/update.

**Comment by elliVM** [comment]
merged into dpl2
